From 15bb53231a42240baf882baeeb265b5bee861190 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=A0=D1=83=D1=81=D0=BB=D0=B0=D0=BD=20=D0=98=D0=B6=D0=B1?=
 =?UTF-8?q?=D1=83=D0=BB=D0=B0=D1=82=D0=BE=D0=B2?= <lrn1986@gmail.com>
Date: Tue, 14 Apr 2015 10:48:26 +0000
Subject: [PATCH 14/20] [W32][ANGLE] Silence two warnings in ANGLE

../webkitgtk-2.4.8/Source/ThirdParty/ANGLE/src/compiler/ParseContext.cpp:538:123: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (op == EOpConstructStruct && !type->isArray() && int(type->getStruct()->fields().size()) != function.getParamCount()) {
                                                                                                                           ^
../webkitgtk-2.4.8/Source/ThirdParty/ANGLE/src/compiler/depgraph/DependencyGraph.cpp:7:0: warning: ignoring #pragma warning  [-Wunknown-pragmas]
 #pragma warning(disable: 4718)
 ^
---
 Source/ThirdParty/ANGLE/src/compiler/ParseContext.cpp             | 2 +-
 Source/ThirdParty/ANGLE/src/compiler/depgraph/DependencyGraph.cpp | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/Source/ThirdParty/ANGLE/src/compiler/ParseContext.cpp b/Source/ThirdParty/ANGLE/src/compiler/ParseContext.cpp
index 9a27952..a93d286 100644
--- a/Source/ThirdParty/ANGLE/src/compiler/ParseContext.cpp
+++ b/Source/ThirdParty/ANGLE/src/compiler/ParseContext.cpp
@@ -535,7 +535,7 @@ bool TParseContext::constructorErrorCheck(const TSourceLoc& line, TIntermNode* n
         return true;
     }
     
-    if (op == EOpConstructStruct && !type->isArray() && int(type->getStruct()->fields().size()) != function.getParamCount()) {
+    if (op == EOpConstructStruct && !type->isArray() && int(type->getStruct()->fields().size()) != int(function.getParamCount())) {
         error(line, "Number of constructor parameters does not match the number of structure fields", "constructor");
         return true;
     }
diff --git a/Source/ThirdParty/ANGLE/src/compiler/depgraph/DependencyGraph.cpp b/Source/ThirdParty/ANGLE/src/compiler/depgraph/DependencyGraph.cpp
index ca661d6..3d306a3 100644
--- a/Source/ThirdParty/ANGLE/src/compiler/depgraph/DependencyGraph.cpp
+++ b/Source/ThirdParty/ANGLE/src/compiler/depgraph/DependencyGraph.cpp
@@ -4,7 +4,9 @@
 // found in the LICENSE file.
 //
 
+#ifdef _MSC_VER
 #pragma warning(disable: 4718)
+#endif
 
 #include "compiler/depgraph/DependencyGraph.h"
 #include "compiler/depgraph/DependencyGraphBuilder.h"
-- 
1.8.5.3

